crates/vsock/src/main.rs

```diff
@@ -287,116 +287,182 @@ mod tests {
     }
 
     #[test]
-    #[serial]
     fn test_vsock_config_setup() {
-        let args = VsockArgs::from_args(3, "/tmp/vhost4.socket", "/tmp/vm4.vsock", 64 * 1024);
+        let test_dir = tempdir().expect("Could not create a temp test directory.");
+        std::env::set_current_dir(&test_dir).expect("Could not chdir to temp test directory.");
 
-        let configs = Vec::<VsockConfig>::try_from(args);
-        assert!(configs.is_ok());
+        {
+            let socket_path = test_dir.path().join("vhost4.socket").display().to_string();
+            let uds_path = test_dir.path().join("vm4.vsock").display().to_string();
+            let args = VsockArgs::from_args(3, &socket_path, &uds_path, 64 * 1024);
 
-        let configs = configs.unwrap();
-        assert_eq!(configs.len(), 1);
+            let configs = Vec::<VsockConfig>::try_from(args);
+            assert!(configs.is_ok());
 
-        let config = &configs[0];
-        assert_eq!(config.get_guest_cid(), 3);
-        assert_eq!(config.get_socket_path(), "/tmp/vhost4.socket");
-        assert_eq!(config.get_uds_path(), "/tmp/vm4.vsock");
-        assert_eq!(config.get_tx_buffer_size(), 64 * 1024);
+            let configs = configs.unwrap();
+            assert_eq!(configs.len(), 1);
+
+            let config = &configs[0];
+            assert_eq!(config.get_guest_cid(), 3);
+            assert_eq!(config.get_socket_path(), socket_path);
+            assert_eq!(config.get_uds_path(), uds_path);
+            assert_eq!(config.get_tx_buffer_size(), 64 * 1024);
+        }
+
+        _ = test_dir.close();
```

Comment:
If we ignore the result anyway, is there any point in explicitly calling `.close()` here? If I understand it correctly, that only helps to assert that cleanup actually happened, but if we ignore the result, we do not do that? Does that maybe also allow to get rid of the extra nested scope?

Reply:
It's there to prevent future rewrites from accidentally dropping `test_dir` earlier than it should. (The compiler would show "test_dir is already borrowed" or "has moved")

As for the error path, now that I think of it it should not ignore the `close()` error.